typescript/src/lib/bitcoin/ecdsa-key.ts

```diff
@@ -0,0 +1,84 @@
+import { BigNumber } from "ethers"
+import { Hex } from "../utils"
+import { ECPairFactory, ECPairInterface } from "ecpair"
+import * as tinysecp from "tiny-secp256k1"
+import { BitcoinNetwork, toBitcoinJsLibNetwork } from "./network"
+
+/**
+ * Checks whether given public key is a compressed Bitcoin public key.
+ * @param publicKey - Public key that should be checked.
+ * @returns True if the key is a compressed Bitcoin public key, false otherwise.
+ */
+function isCompressedPublicKey(publicKey: string): boolean {
+  // Must have 33 bytes and 02 or 03 prefix.
+  return (
+    publicKey.length == 66 &&
+    (publicKey.substring(0, 2) == "02" || publicKey.substring(0, 2) == "03")
+  )
+}
+
+/**
+ * Compresses the given uncompressed Bitcoin public key.
+ * @param publicKey Uncompressed 64-byte public key as an unprefixed hex string.
+ * @returns Compressed 33-byte public key prefixed with 02 or 03.
+ */
+function compressPublicKey(publicKey: string | Hex): string {
+  if (typeof publicKey === "string") {
```

Comment:
We should make `Hex.from` accept also `Hex` to make things easier and don't need to check type here.

Reply:
Yeah but not as part of this PR. Something for the future.